crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -350,630 +380,627 @@ impl Instruction {
+                return binary.evaluate(ctx, self.id, self.res_type, eval_fn)
+            }
+            Operation::Cast(value) => {
+                if let Some(l_const) = ctx.get_const_value(*value) {
+                    if self.res_type == ObjectType::NativeField {
+                        return NodeEval::Const(FieldElement::from(l_const), self.res_type);
+                    }
+                    return NodeEval::Const(
+                        FieldElement::from(l_const % (1_u128 << self.res_type.bits())),
+                        self.res_type,
+                    );
+                }
             }
-            ObjectType::Signed(b) | ObjectType::Unsigned(b) => {
-                assert!(b < 128); //we do not support integers bigger than 128 bits for now.
-                (c.to_u128(), b)
-            } //TODO check how to handle signed integers
-            _ => todo!(),
+            Operation::Not(value) => {
+                if let Some(l_const) = ctx.get_const_value(*value) {
+                    return NodeEval::Const(FieldElement::from(!l_const), self.res_type);
+                }
+            }
+            Operation::Phi { .. } => (), //Phi are simplified by simply_phi() later on; they must not be simplified here
+            _ => (),
         }
+        NodeEval::VarOrInstruction(self.id)
     }
 
-    pub fn node_evaluate(n: &NodeEval) -> (bool, Option<u128>, u32) {
-        match n {
-            &NodeEval::Const(c, t) => {
-                let cv = Instruction::get_const_value(c, t);
-                (c.is_zero(), Some(cv.0), cv.1)
+    // Simplifies trivial Phi instructions by returning:
+    // None, if the instruction is unreachable or in the root block and can be safely deleted
+    // Some(id), if the instruction can be replaced by the node id
+    // Some(ins_id), if the instruction is not trivial
+    pub fn simplify_phi(ins_id: NodeId, phi_arguments: &[(NodeId, BlockId)]) -> Option<NodeId> {
+        let mut same = None;
+        for op in phi_arguments {
+            if Some(op.0) == same || op.0 == ins_id {
+                continue;
             }
-            _ => (false, None, 0),
+            if same.is_some() {
+                //no simplification
+                return Some(ins_id);
+            }
+
+            same = Some(op.0);
         }
+        //if same.is_none()  => unreachable phi or in root block, can be replaced by ins.lhs (i.e the root) then.
+        same
     }
 
-    //Evaluate the instruction value when its operands are constant (constant folding)
-    pub fn evaluate(&self, lhs: &NodeEval, rhs: &NodeEval) -> NodeEval {
-        //let mut l_sign = false; //TODO
-        let (l_is_zero, l_constant, l_bsize) = Instruction::node_evaluate(lhs);
-        let (r_is_zero, r_constant, r_bsize) = Instruction::node_evaluate(rhs);
-        let r_is_const = r_constant.is_some();
-        let l_is_const = l_constant.is_some();
+    pub fn standard_form(&mut self) {
```

Comment:
`standard_form` no longer converts `>` into `<` or `<=` into `>=` - instead we simply do not support these opcodes to begin with which statically prevents them. Instead, when we create the ssa for a `>` binary op we create a `<` instruction from the start.